### PR TITLE
rubysrc2cpg: fix multiple assignments

### PR DESCRIPTION
**Pull request by tuxology**
Fixes issues where multiple RHS/LHS values in assignments (eg. c, d = a, b) work properly with dataflows. This fixes the previously broken test case where flow from a to c was not working:

```
a = 1
b = 2
c, d = a, b
puts c
```

Also works with other cases where RHS values are in array. eg. a, b = [1, 2]. More complex cases will be tested and added in subsequent PRs

### CODE REVIEW

**Comment by tuxology** [comment]
I reshaped RHS ast for multiple assignments to remove arrayInitializer call. However, we are now missing everything except first element on RHS 😢 

As an example, for following code, 

```
a, b, c = [1, 2, 3]
a, b, c = b, c, a
str1, str2 = ["hello", "world"]
```

we have this AST 👇 

image

⚠️ **Note:** Function calls on RHS are still not modeled.

**Review by rahul-privado — APPROVED** [review]

**Comment by tuxology** [comment]
6 more tests failing. Need to work on this more

**Review by rahul-privado — CHANGES_REQUESTED** [review]
UT fixes needed

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala [hunk 1]
```diff
@@ -1461,16 +1439,46 @@ class AstCreator(filename: String, global: Global)
...
-      .code(ctx.getText)
-      .methodFullName(operatorName)
-      .signature("")
-      .dispatchType(DispatchTypes.STATIC_DISPATCH)
-      .typeFullName(Defines.Any)
-      .lineNumber(ctx.EQ().getSymbol().getLine())
-      .columnNumber(ctx.EQ().getSymbol().getCharPositionInLine())
-    Seq(callAst(callNode, lhsAsts ++ rhsAsts))
+
+    /* If we get anything else than Identifier or Literal, we should ignore and
```
Comment (fabsx00) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:1443]: `If we get anything other than...`

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala [hunk 2]
```diff
@@ -1461,16 +1439,46 @@ class AstCreator(filename: String, global: Global)
...
+      }
+
+      val rhsCode = argPair._2.nodes.headOption match {
+        case Some(id: NewIdentifier) => id.code
+        case Some(lit: NewLiteral)   => lit.code
+        case _                       => ""
+      }
+
+      val syntheticCallNode = NewCall()
+        .name(operatorName)
```
Comment (fabsx00) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:1471]: `nameExact`, `codeExact` (where ever possible)

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala [hunk 3]
```diff
@@ -1461,16 +1439,46 @@ class AstCreator(filename: String, global: Global)
...
+        .code(lhsCode + " = " + rhsCode)
+        .methodFullName(operatorName)
+        .signature("")
```
Comment (fabsx00) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:1474]: This line can/should be removed.

**Review by fabsx00 — APPROVED** [review]

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala (hunk 2 above)
Comment (tuxology) [reply joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:1471]: `NewCall()` only allows `code` and `name` values to be added

**Pull request merged**